src/algebra/group_with_zero/defs.lean

```diff
@@ -74,6 +74,12 @@ cancel_monoid_with_zero.mul_left_cancel_of_ne_zero ha h
 lemma mul_right_cancel' (hb : b ≠ 0) (h : a * b = c * b) : a = c :=
 cancel_monoid_with_zero.mul_right_cancel_of_ne_zero hb h
 
+lemma mul_right_injective' (ha : a ≠ 0) : function.injective (λ b, a * b) :=
+λ b c, mul_left_cancel' ha
+
+lemma mul_left_injective' (hb : b ≠ 0) : function.injective (λ a, a * b) :=
```

Comment:
Note that `mul_left_injective` is stated as the following, which is syntactically more useful as `(mul_left_injective' _).eq_iff` doesn't produce something that needs delta unfolding.
```suggestion
lemma mul_left_injective' (hb : b ≠ 0) : function.injective ((*) a) :=
```

Of course, the right version unavoidably needs delta unfolding, so perhaps this isn't worth worrying about; but if so, then we should probably change the unprimed version to also not worry about it.

Reply:
I recall that there was a preference for `λ`s over operator sections, but I can't find the relevant Zulip thread anymore. Since the neatness of `.eq_iff` is a good argument, I'll change it over.